src/tests/test_alerts.py

```diff
@@ -16,43 +43,43 @@
      "is_acknowledged": False, "created_at": "2020-11-03T11:18:45.447773"},
 ]
 
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%f"
 
-async def mock_fetch_ongoing_alerts(table, query_filters=None, excluded_events_filter=None):
-    excluded_events = []
-    for entry in table:
-        if all(entry[k] == v for k, v in excluded_events_filter.items()):
-            excluded_events.append(entry["event_id"])
 
-    response = []
-    for entry in table:
-        if all(entry[k] == v for k, v in query_filters.items()):
-            if entry["event_id"] not in excluded_events:
-                response.append(entry)
-    return response
+def update_only_datetime(entity_as_dict):
+    to_return = entity_as_dict.copy()
+    if "created_at" in to_return:
+        to_return["created_at"] = datetime.strptime(to_return["created_at"], DATETIME_FORMAT)
+    return to_return
 
 
-def _patch_session(monkeypatch, mock_table):
-    # DB patching
-    monkeypatch.setattr(alerts, "alerts", mock_table)
-    # Sterilize all DB interactions through CRUD override
-    monkeypatch.setattr(crud, "get", pytest.mock_get)
-    monkeypatch.setattr(crud, "fetch_one", pytest.mock_fetch_one)
-    monkeypatch.setattr(crud, "fetch_all", pytest.mock_fetch_all)
-    monkeypatch.setattr(crud, "fetch_ongoing_alerts", mock_fetch_ongoing_alerts)
-    monkeypatch.setattr(crud, "post", pytest.mock_post)
-    monkeypatch.setattr(crud, "put", pytest.mock_put)
-    monkeypatch.setattr(crud, "delete", pytest.mock_delete)
+ACCESS_TABLE_FOR_DB = list(map(update_only_datetime, ACCESS_TABLE))
+USER_TABLE_FOR_DB = list(map(update_only_datetime, USER_TABLE))
+DEVICE_TABLE_FOR_DB = list(map(update_only_datetime, DEVICE_TABLE))
+EVENT_TABLE_FOR_DB = list(map(update_only_datetime, EVENT_TABLE))
+ALERT_TABLE_FOR_DB = list(map(update_only_datetime, ALERT_TABLE))
+MEDIA_TABLE_FOR_DB = list(map(update_only_datetime, MEDIA_TABLE))
 
 
-def test_get_alert(test_app, monkeypatch):
+async def init_test_db(monkeypatch, test_db):
+    monkeypatch.setattr(crud, "database", test_db)
+    await populate_db(test_db, db.accesses, ACCESS_TABLE_FOR_DB)
+    await populate_db(test_db, db.users, USER_TABLE_FOR_DB)
 
-    # Sterilize DB interactions
-    mock_alert_table = deepcopy(ALERT_TABLE)
-    _patch_session(monkeypatch, mock_alert_table)
+    await populate_db(test_db, db.devices, DEVICE_TABLE_FOR_DB)
+    await populate_db(test_db, db.events, EVENT_TABLE_FOR_DB)
+    await populate_db(test_db, db.alerts, ALERT_TABLE_FOR_DB)
+    await populate_db(test_db, db.media, MEDIA_TABLE_FOR_DB)
```

Comment:
Mmmmh, patching all tables while only two routes require more than the access table might be an overkill (and it helps raising error if it interacts with the table while it shouldn't)

Perhaps we could have a more flexible patching?

Reply:
copypasted: Those are not patches per se, but because of the foreign keys constraints, I can't populate a DB referring another DB if the foreign entities don't exist. And so on, hence why I need to populate everything here.